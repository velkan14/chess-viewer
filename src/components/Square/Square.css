.Square {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #eeeed2;
  color: #769656;
  position: relative;
}

.black {
  background-color: #769656;
  color: #eeeed2;
}

.numberTag {
  position: absolute;
  top: 0.1rem;
  left: 0.1rem;
  font-size: 1rem;
}

.letterTag {
  position: absolute;
  bottom: 0.1rem;
  right: 0.1rem;
  font-size: 1rem;
}

.target {
  border: 2px dashed firebrick;
}
