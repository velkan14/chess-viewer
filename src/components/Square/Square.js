import React from "react";
import classNames from "classnames";
import "./Square.css";

const Square = (props) => {
  const { color, letterTag, numberTag, target } = props;
  return (
    <div
      className={classNames("Square", {
        black: color === "black",
        target: target,
      })}
    >
      {numberTag && (
        <span className="numberTag" data-testid="numberTag">
          {numberTag}
        </span>
      )}
      {letterTag && (
        <span className="letterTag" data-testid="letterTag">
          {letterTag}
        </span>
      )}
      {props.children}
    </div>
  );
};

export default Square;
