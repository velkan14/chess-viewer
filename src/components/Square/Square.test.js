import React from "react";
import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import shallowToJson from "enzyme-to-json";
import Square from "./Square";

test("Render tags", () => {
  const { getByTestId } = render(
    <Square color="white" letterTag="a" numberTag="1" target={false} />
  );
  expect(getByTestId(/letterTag/i).textContent).toBe("a");
  expect(getByTestId(/numberTag/i).textContent).toBe("1");
});

test("Doesn't render tags", () => {
  const { queryByTestId } = render(<Square color="white" target={false} />);
  expect(queryByTestId(/letterTag/i)).toBeNull();
  expect(queryByTestId(/numberTag/i)).toBeNull();
});

test("Doesn't render tags", () => {
  const { queryByTestId } = render(<Square color="white" target={false} />);
  expect(queryByTestId(/letterTag/i)).toBeNull();
  expect(queryByTestId(/numberTag/i)).toBeNull();
});

test("Test snapshots", () => {
  let wrapper = shallow(
    <Square color="white" letterTag="a" numberTag="1" target={false} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();

  wrapper = shallow(
    <Square color="black" letterTag="a" numberTag="1" target={false} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();

  wrapper = shallow(<Square color="white" numberTag="1" target={false} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();

  wrapper = shallow(<Square color="white" letterTag="a" target={false} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();

  wrapper = shallow(<Square color="white" target={false} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();

  wrapper = shallow(
    <Square color="white" letterTag="a" numberTag="1" target={true} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
