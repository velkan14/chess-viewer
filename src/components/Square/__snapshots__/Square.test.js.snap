// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshots 1`] = `
<div
  className="Square"
>
  <span
    className="numberTag"
    data-testid="numberTag"
  >
    1
  </span>
  <span
    className="letterTag"
    data-testid="letterTag"
  >
    a
  </span>
</div>
`;

exports[`Test snapshots 2`] = `
<div
  className="Square black"
>
  <span
    className="numberTag"
    data-testid="numberTag"
  >
    1
  </span>
  <span
    className="letterTag"
    data-testid="letterTag"
  >
    a
  </span>
</div>
`;

exports[`Test snapshots 3`] = `
<div
  className="Square"
>
  <span
    className="numberTag"
    data-testid="numberTag"
  >
    1
  </span>
</div>
`;

exports[`Test snapshots 4`] = `
<div
  className="Square"
>
  <span
    className="letterTag"
    data-testid="letterTag"
  >
    a
  </span>
</div>
`;

exports[`Test snapshots 5`] = `
<div
  className="Square"
/>
`;

exports[`Test snapshots 6`] = `
<div
  className="Square target"
>
  <span
    className="numberTag"
    data-testid="numberTag"
  >
    1
  </span>
  <span
    className="letterTag"
    data-testid="letterTag"
  >
    a
  </span>
</div>
`;
